### PR TITLE
feat: add service registry

### LINKED ISSUE #334: [Config] Register `arana` as compute service node to Registry center.
**What would you like to be added**:

**Why is this needed**:

### PR DESCRIPTION
**Pull request by georgehao**
**What this PR does**:
add registry

**Which issue(s) this PR fixes**:

Fixes #334

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/boot/discovery.go [hunk 1]
```diff
@@ -282,6 +282,10 @@ func (fp *discovery) ListListeners(ctx context.Context) []*config.Listener {
 	return fp.options.Listeners
 }
 
+func (fp *discovery) ListServiceRegistry(ctx context.Context) *config.Registry {
```
Comment (chuntaojun) [inline pkg/boot/discovery.go:285]: ListServiceRegistry => GetServiceRegistry is better ?

File: conf/bootstrap.yaml [hunk 2]
```diff
@@ -23,6 +23,23 @@ listeners:
     socket_address:
       address: 0.0.0.0
       port: 13306
+
+registry:
+    name: etcd
+    root_path: arana
+    options:
+      endpoints: "http://127.0.0.1:2379"
```
Comment (jjeffcaii) [inline conf/bootstrap.yaml:31]: How about using existing config directly?

File: pkg/registry/store/etcdv3.go [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package store
+
+import (
+	"context"
+	"github.com/arana-db/arana/pkg/util/log"
```
Comment (dongzl) [inline pkg/registry/store/etcdv3.go:22]: format import

**Review by dongzl — CHANGES_REQUESTED** [review]

**Review by jjeffcaii — APPROVED** [review]

File: cmd/start/start.go [hunk 4]
```diff
@@ -92,6 +93,21 @@ func Run(bootstrapConfigPath string) {
 		propeller.AddListener(listener)
 	}
 
+	// init service registry
+	registryConf := discovery.GetServiceRegistry(context.Background())
+	if registryConf != nil && registryConf.Enable {
+		serviceRegistry, err := registry.InitRegistry(registryConf)
+		if err != nil {
+			log.Fatalf("create service registry failed: %v", err)
```
Comment (jjeffcaii) [inline cmd/start/start.go:101]: How about use ERR level? The `Fatalf` will cause process exiting, but here I think it's not necessary to let process killed.

File: cmd/start/start.go [hunk 5]
```diff
@@ -92,6 +93,21 @@ func Run(bootstrapConfigPath string) {
...
+			return
+		}
+
+		if err := registry.DoRegistry(context.Background(), serviceRegistry, "service", listenersConf); err != nil {
+			log.Fatalf("do service register failed: %v", err)
```
Comment (jjeffcaii) [inline cmd/start/start.go:106]: Same as above.

**Review by dongzl — APPROVED** [review]

**Review by chuntaojun — APPROVED** [review]

**Pull request merged**